pkg/sql/plan/build_util.go

```diff
@@ -165,6 +166,42 @@ func buildDefaultExpr(col *tree.ColumnTableDef, typ *plan.Type) (*plan.Default,
 	}, nil
 }
 
+func buildOnUpdate(col *tree.ColumnTableDef, typ *plan.Type) (*plan.Expr, error) {
+	var expr tree.Expr = nil
+
+	for _, attr := range col.Attributes {
+		if s, ok := attr.(*tree.AttributeOnUpdate); ok {
+			expr = s.Expr
+			break
+		}
+	}
+
+	if expr == nil {
+		return nil, nil
+	}
+
+	binder := NewDefaultBinder(nil, nil, typ)
+	planExpr, err := binder.BindExpr(expr, 0, false)
+	if err != nil {
+		return nil, err
+	}
+
+	onUpdateExpr, err := makePlan2CastExpr(planExpr, typ)
+	if err != nil {
+		return nil, err
+	}
+
+	// try to calculate on update value, return err if fails
+	bat := batch.NewWithSize(0)
+	bat.Zs = []int64{1}
```

Comment:
This Zs is so strange.   Is this the right way to do things?   Should we init in within NewWithSize?

Reply:
Here it simply means that a row of data exists